sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/util/BuilderHelper.java

```diff
@@ -109,6 +130,18 @@ public static String getEndpoint(BlobUrlParts parts) {
         }
     }
 
+    /**
+     * Validates that the client is properly configured for using cpk.
+     *
+     * @param customerProvidedKey The cpk object.
+     * @param endpoint The endpoint for the client.
+     */
+    public static void validateCpk(CpkInfo customerProvidedKey, String endpoint) {
```

Comment:
Just a thought, instead of failing why dont we change their url to https and log a warning or something?